developer/provider/windows-powershell-provider-quickstart.md

```diff
@@ -1,167 +1,174 @@
+      if (String.IsNullOrEmpty(drive.Root) || (File.Exists(drive.Root) == false))
+      {
+        WriteError(new ErrorRecord(
+                   new ArgumentException("drive.Root"),
+                   "NoRoot",
+                   ErrorCategory.InvalidArgument,
+                   drive));
+
+        return null;
+      }
+
+      // Create a new drive and create an ODBC connection to the new drive.
+      AccessDBPSDriveInfo accessDBPSDriveInfo = new AccessDBPSDriveInfo(drive);
+      OdbcConnectionStringBuilder builder = new OdbcConnectionStringBuilder();
+
+      builder.Driver = "Microsoft Access Driver (*.mdb)";
+      builder.Add("DBQ", drive.Root);
+
+      OdbcConnection conn = new OdbcConnection(builder.ConnectionString);
+      conn.Open();
+      accessDBPSDriveInfo.Connection = conn;
+
+      return accessDBPSDriveInfo;
+    }
+```
+
+ The following is the AccessDBPSDriveInfo internal class that includes the constructor used to create a new drive, and contains the state information for the drive.
+
+```csharp
+internal class AccessDBPSDriveInfo : PSDriveInfo
+  {
+    /// <summary>
+    /// A reference to the connection to the database.
+    /// </summary>
+    private OdbcConnection connection;
+
+    /// <summary>
+    /// Initializes a new instance of the AccessDBPSDriveInfo class.
+    /// The constructor takes a single argument.
+    /// </summary>
+    /// <param name="driveInfo">Drive defined by this provider</param>
+    public AccessDBPSDriveInfo(PSDriveInfo driveInfo)
+           : base(driveInfo)
+    {
+    }
+
+    /// <summary>
+    /// Gets or sets the ODBC connection information.
+    /// </summary>
+    public OdbcConnection Connection
+    {
+        get { return this.connection; }
+        set { this.connection = value; }
+    }
+  }
+```
+
+### Implementing RemoveDrive
+
+ The [System.Management.Automation.Provider.Drivecmdletprovider.Removedrive*](/dotnet/api/System.Management.Automation.Provider.DriveCmdletProvider.RemoveDrive) method is called by the Windows PowerShell engine when a user calls the [Microsoft.Powershell.Commands.Remove-Psdrive](/dotnet/api/Microsoft.PowerShell.Commands.Remove-PSDrive) cmdlet. The method in this provider closes the connection to the Access database.
+
+```csharp
+protected override PSDriveInfo RemoveDrive(PSDriveInfo drive)
+    {
+      // Check if drive object is null.
+      if (drive == null)
+      {
+        WriteError(new ErrorRecord(
+                   new ArgumentNullException("drive"),
+                   "NullDrive",
+                   ErrorCategory.InvalidArgument,
+                   drive));
+
+        return null;
+      }
+
+      // Close the ODBC connection to the drive.
+      AccessDBPSDriveInfo accessDBPSDriveInfo = drive as AccessDBPSDriveInfo;
+
+      if (accessDBPSDriveInfo == null)
+      {
+         return null;
+      }
+
+      accessDBPSDriveInfo.Connection.Close();
+
+      return accessDBPSDriveInfo;
+    }
+```


```

Comment:
Control character

Reply:
This was an issue with my code fence logic and is fixed